service/history/api/get_history_util.go

```diff
@@ -139,6 +141,24 @@ func GetHistory(
 			tag.Error(err),
 		)
 		return nil, nil, err
+	case *serialization.DeserializationError:
+		// log event
+		shard.GetLogger().Error("encountered de-serialization error",
+			tag.WorkflowNamespaceID(namespaceID.String()),
+			tag.WorkflowID(execution.GetWorkflowId()),
+			tag.WorkflowRunID(execution.GetRunId()),
+			tag.Error(err),
+		)
+		return nil, nil, err
+	case *serialization.SerializationError:
```

Comment:
nit: merge all 3 into one `case`. The message can just be `encountered data corruption error`, the actual error is tagged anyway.